tests/core/fs/storage/test_workspace_storage.py

```diff
@@ -19,13 +20,14 @@
 )
 
 
-def create_manifest(device, type=LocalWorkspaceManifest):
+def create_manifest(device, type=LocalWorkspaceManifest, legacy=False):
+    author = None if legacy else device.device_id
```

Comment:
a comment about what this legacy option is about would be interesting here (and maybe even rename `legacy` into `legacy_base_manifest_author_is_none`)